packages/gatsby-plugin-image/src/components/hooks.ts

```diff
@@ -27,8 +27,8 @@ export type IGatsbyImageDataParent<T = never> = T & {
 export type IGatsbyImageParent<T = never> = T & {
   gatsbyImage: IGatsbyImageData
 }
-export type FileNode = Node & {
-  childImageSharp?: IGatsbyImageDataParent<Node>
+export type FileNode = Partial<Node> & {
+  childImageSharp?: IGatsbyImageDataParent<Partial<Node>>
```

Comment:
It feels like we don't care about this looking like a "Node" properties at all and just care about `childImageSharp`? maybe we drop `Node` completely and have just have object type with `childImageSharp` optional field?